src/components/GraphiQL.js

```diff
@@ -694,6 +710,42 @@ export class GraphiQL extends React.Component {
     editor.setValue(print(parse(editor.getValue())));
   };
 
+  handleDefragQuery = () => {
+    const editor = this.getQueryEditor();
+    const ast = parse(editor.getValue());
+
+    let operations = ast.definitions.filter(elem => {
+      return elem.kind !== 'FragmentDefinition';
+    });
+
+    const fragments = {};
+    ast.definitions
+      .filter(elem => {
+        return elem.kind === 'FragmentDefinition';
+      })
+      .forEach(frag => {
+        frag.kind = 'InlineFragment';
+        fragments[frag.name.value] = frag;
+      });
+    operations = operations.map(op => this.resolveDefinition(fragments, op));
+
+    ast.definitions = operations;
+    editor.setValue(print(ast));
+  };
+
+  resolveDefinition = (fragments, obj) => {
+    let definition = obj;
+    if (definition.kind === 'FragmentSpread') {
```

Comment:
I think we have an enum value for `FragmentSpread` and keywords like these above. Let's use that instead